pkg/storageos/scheduler_extender.go

```diff
@@ -70,6 +70,9 @@ func (s Deployment) createSchedulerDeployment(replicas int32) error {
 		},
 	}
 
+	// Add cluster config tolerations.
+	s.addTolerations(&spec.Template.Spec)
```

Comment:
This adds the StorageOSCluster toleration to the kube-scheduler deployment. Since the k8s node in e2e test is tainted, without this toleration in kube-scheduler deployment, scheduling of kube-scheduler pod fails. Since the admission controller mutates any StorageOS volume consumer pod to use the scheduler and the scheduler isn't running, the consumer pods, including NFS Server pods fail.